akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Blueprint.scala

```diff
@@ -30,8 +31,13 @@ private[http2] final case class Http2SubStream(initialHeaders: ParsedHeadersFram
 
 object Http2Blueprint {
   // format: OFF
-  def serverStack(): BidiFlow[HttpResponse, ByteString, ByteString, HttpRequest, NotUsed] = {
-    httpLayer() atop
+  def serverStack(settings: ServerSettings, log: LoggingAdapter): BidiFlow[HttpResponse, ByteString, ByteString, HttpRequest, NotUsed] = {
+    val parserSettings = settings.parserSettings
+    val httpHeaderParser = HttpHeaderParser.prime(HttpHeaderParser.unprimed(parserSettings, log, { info =>
```

Comment:
Will refactor that. Didn't notice that `HttpHeaderParser` is not thread-safe. Pity, as otherwise already parsed headers could be re-used for the lifetime of the server. Another reason to rewrite in future - will add a comment.